features/steps/method_summary.py

```diff
@@ -48,13 +48,19 @@ def check_method_has_github_resources(context, method_name):
     WebDriverWait(driver, timeout=timeout).until(
         lambda d: d.find_element(By.PARTIAL_LINK_TEXT, value=method_spec_link_text)
     )
-    WebDriverWait(driver, timeout=timeout).until(
-        lambda d: d.find_element(By.PARTIAL_LINK_TEXT, value=code_link_text)
-    )
+
     WebDriverWait(driver, timeout=timeout).until(
         lambda d: d.find_element(By.PARTIAL_LINK_TEXT, value=user_docs_link_text)
     )
 
+    code_link = (
+        WebDriverWait(driver, timeout=timeout)
+        .until(lambda d: d.find_element(By.PARTIAL_LINK_TEXT, value=code_link_text))
+        .get_attribute("href")
+    )
+
+    assert code_link.endswith(".py")
```

Comment:
Minor - the comment further up "asserts are not needed here" could be amended since we are now introducing asserts. Something like, testing of links without asserts will fail due to timeout.